python/python_direct/ops.cpp

```diff
@@ -2303,6 +2303,55 @@ Returns
 TensorView
     The selected tensor.
 )");
+  ops.def(
+      "scatter",
+      [](TensorView* arg1, TensorView* index, TensorView* src, int64_t dim)
+          -> TensorView* {
+        NVF_CHECK(
+            arg1->nDims() == index->nDims() && arg1->nDims() == src->nDims(),
+            "Tensor arguments have different dimensions ",
+            arg1->nDims(),
+            ", ",
+            index->nDims(),
+            " and ",
+            src->nDims());
+        auto num_dims = (int64_t)arg1->nDims();
+        NVF_CHECK(
+            dim >= -num_dims && dim < num_dims,
+            "Tensor arguments have dimension ",
+            num_dims,
+            " so dim argument must satisfy ",
+            -num_dims,
+            " <= dim < ",
+            num_dims,
+            ", but received ",
+            dim);
```

Comment:
this is checked by `wrapDim` inside scatter already. With direct binding, I don't think we need to validate stuff before the c++ api.